ucp/_libs/ucp_py.pyx

```diff
@@ -233,8 +233,8 @@ cdef class Endpoint:
     def connect(self, ip, port):
         self.ep = ucp_py_get_ep(ip, port)
         if <void *> NULL == self.ep:
-            raise NameError('Failed to connect to remote endpoint')
-        return
+            return -1
+        return 0
```

Comment:
Python conventions would be to return True if success and False if failure (or raise an OSError).  Not a big deal though.  I think that any Cython dev will see this for C style conventions and be fine.